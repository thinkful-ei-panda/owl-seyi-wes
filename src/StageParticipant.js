import React from 'react';

function StageParticipant(props){
    if (props.participant.onStage===true){
        return  (<div className='stage-participant-container'>                    
                    <div>
                        <div className='stage-name'>{props.participant.name}</div>
                    </div>
                    <div>
                        <img src={props.participant.avatar} alt='avatar' />
                    </div>
                </div>)
    } else{
        return <></>
    }
}

export default StageParticipant