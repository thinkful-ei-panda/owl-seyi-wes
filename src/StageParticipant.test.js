import React from 'react';
import ReactDOM from 'react-dom';
import StageParticipant from './StageParticipant';
import store from './store'

import renderer from 'react-test-renderer';

//<StageParticipant key={props.participants[index].id} participant={props.participants[index]} />
describe('StageParticipant', () => {

    const props={
        key:0,
        participant:{
            id: 1,
            name: 'Koren Templeton',
            avatar:
                'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
            inSession: true,
            onStage: true
        }
    }
  it('renders without crashing', () => {

    const div = document.createElement('div');
    ReactDOM.render(<StageParticipant {...props}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<StageParticipant {...props}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

});

