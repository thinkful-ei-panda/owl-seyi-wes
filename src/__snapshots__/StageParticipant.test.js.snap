// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StageParticipant renders the UI as expected 1`] = `
<div
  className="stage-participant-container"
>
  <div>
    <div
      className="stage-name"
    >
      Koren Templeton
    </div>
  </div>
  <div>
    <img
      alt="avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
  </div>
</div>
`;
